docs/docsite/rst/guide_aap.rst

```diff
@@ -0,0 +1,165 @@
+Using cloud.terraform in Ansible Automation Platform
+*************************************************************
+
+The ``cloud.terraform`` collection requires some extra configuration to run in `Ansible Automation Platform (AAP) <https://www.redhat.com/en/technologies/management/ansible>`_; in particular the default execution environment in AAP does not include a Terraform binary due to `Terraform licensing restrictions <https://www.hashicorp.com/license-faq>`_. You must create an execution environment containing a Terraform binary in order to run any Terraform commands from within AAP.
+
+In addition, a remote Terraform backend should be used to store Terraform state as any local state files created in AAP jobs are ephemeral. The Terraform backend credential type allows secure storage of remote backend configuration to enable this.
+
+=====================
+Execution Environment
+=====================
+
+Job templates and inventory source syncs using the ``cloud.terraform`` collection in AAP must be run in an execution environment containing a Terraform binary. You can create one using `ansible-builder <https://ansible.readthedocs.io/projects/builder/en/latest/#>`_ following the instructions below.
+
+Instructions
+------------
+
+1. Install ansible-builder:
+
+.. code-block:: console
+
+    pip install ansible-builder
+
+2. Create an execution-environment.yml configuration similar to this example. Note the inclusion of the ``cloud.terraform`` collection and the ``additional_build_steps`` section that downloads and installs Terraform.
+
+.. code-block:: yaml
+
+    ---
+    version: 3
+
+    images:
+    base_image:
+      name: quay.io/centos/centos:stream9
+
+    dependencies:
+    ansible_core:
+      package_pip: ansible-core
+    ansible_runner:
+      package_pip: ansible-runner
+    galaxy:
+      collections:
+        - name: cloud.terraform
+
+    additional_build_steps:
+    append_base: |
+      RUN yum install -y git
+      RUN curl https://rpm.releases.hashicorp.com/RHEL/hashicorp.repo | tee /etc/yum.repos.d/terraform.repo
+      RUN yum install -y terraform
+
+3. Build the execution environment image:
+
+.. code-block:: console
+
+    ansible-builder build
+
+The Containerfile/Dockerfile and build files should now be in ``/context`` and you can see your newly created image in podman or docker:
+
+.. code-block:: console
+
+    podman images
+
+    REPOSITORY                       TAG         IMAGE ID      CREATED         SIZE
+    localhost/ansible-execution-env  latest      2976756bc5f3  5 seconds ago   521 MB
+    <none>                           <none>      74ac0ff084f1  14 seconds ago  523 MB
+    <none>                           <none>      2498e29966ee  21 seconds ago  521 MB
+    quay.io/centos/centos            stream9     94714b8dc9d7  2 days ago      168 MB
+
+
+4. Publish the image:
+
+.. code-block:: console
+
+    podman login quay.io
+    podman push localhost/ansible-execution-env quay.io/<username>/<repository>
```

Comment:
Since this is specifically for AAP, they should have access to private automation hub, and that's likely a better option for publishing an image.

Reply:
Updated